Add python script examples

andrew-gardener opened the pull request.

andrew-gardener committed:
Add python script example

Commit:
Updated scripts to use .env file and python-dotenv.
Added .gitignore

craigdsthompson commented:
@andrew-gardener I reviewed your code and have made some changes. Specifically, I think we should recommend using `.env` files to minimize copy/pasting tokens to the command line (either to set environment variables, or to pass as command line parameters). This prevents the token leaking via `ps`.

If you are happy with my changes, I'll go ahead and merge.

andrew-gardener commented on api_scripts/python/robust.py:
```diff
@@ -38,18 +38,26 @@
+    # ensure access token is available
+    TOKEN = os.environ.get('CANVAS_ACCESS_TOKEN')
+    if TOKEN == None:
+        print("No access token found. Please set `CANVAS_ACCESS_TOKEN`")
```
I think it would be better to do 
```
TOKEN = os.environ.get('CANVAS_ACCESS_TOKEN')
while not TOKEN:
    TOKEN = click.prompt("Enter your access token", hide_input=True)
```

craigdsthompson replied:
Good idea. Fixed in fd29f11.

craigdsthompson commented on api_scripts/python/simple_input.py:
```diff
@@ -0,0 +1,71 @@
+    params = {
+        "per_page": str(PER_PAGE),
+        "page": str(page),
+        "sort": sort_by,
```
Does this actually work for you? I tested, and couldn't get sorting to work. It looks to me like the courses endpoint doesn't have a sort param:

https://canvas.instructure.com/doc/api/courses.html#method.courses.index

(same comment applies to all 3 examples)

andrew-gardener replied:
yet, doesn't work. I made the examples by converting a script I did using the accounts course endpoint (which does have sorting). I only have two canvas courses so didn't notice...

I'll add something else to the query since I only really wanted to use sorting as an example of something the user would need to provide

craigdsthompson reviewed (commented):
Looks great in general. I've added a commit to use .env files. What do you think?

Commit:
Prompt the user (repeatedly) for a token if one does not exist
in the environment rather than terminating.

andrew-gardener committed:
Switched sorting to course state in examples

Also downgraded python-dotenv to 0.8.2 so it will work in both python 2.7 and 3.x (find_dotenv was empty for 2.7 for some reason)

andrew-gardener commented:
`.env` is fine. I didn't go that route since I wasn't sure how conferrable some people might be with them (so I went with explicitly setting them) but environment management libraries are always better

Anyways, switch sorting with course state so the examples are still very similar requirement wise.

craigdsthompson commented:
I've tested your latest changes. I'm happy to merge, are you?
